lib/Doctrine/ORM/Persisters/ManyToManyPersister.php

```diff
@@ -37,130 +37,288 @@ class ManyToManyPersister extends AbstractCollectionPersister
+        $this->conn->executeUpdate($this->getDeleteSQL($coll), $this->getDeleteSQLParameters($coll));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update(PersistentCollection $coll)
+    {
+        $mapping = $coll->getMapping();
+
+        if ( ! $mapping['isOwningSide']) {
+            return; // ignore inverse side
         }
 
-        return 'DELETE FROM ' . $tableName
-            . ' WHERE ' . implode(' = ? AND ', $columns) . ' = ?';
+        $diff   = $coll->getDeleteDiff();
+        $sql    = $this->getDeleteRowSQL($coll);
+
+        foreach ($diff as $element) {
+            $this->conn->executeUpdate($sql, $this->getDeleteRowSQLParameters($coll, $element));
+        }
+
+        $diff   = $coll->getInsertDiff();
+        $sql    = $this->getInsertRowSQL($coll);
+
+        foreach ($diff as $element) {
+            $this->conn->executeUpdate($sql, $this->getInsertRowSQLParameters($coll, $element));
+        }
     }
 
     /**
      * {@inheritdoc}
-     *
-     * @override
-     *
-     * @internal Order of the parameters must be the same as the order of the columns in getDeleteRowSql.
      */
-    protected function getDeleteRowSQLParameters(PersistentCollection $coll, $element)
+    public function count(PersistentCollection $coll)
     {
-        return $this->collectJoinTableColumnParameters($coll, $element);
+        $conditions     = array();
+        $params         = array();
+        $mapping        = $coll->getMapping();
+        $association    = $mapping;
+        $class          = $this->em->getClassMetadata($mapping['sourceEntity']);
+        $id             = $this->uow->getEntityIdentifier($coll->getOwner());
+
+        if ( ! $mapping['isOwningSide']) {
+            $targetEntity   = $this->em->getClassMetadata($mapping['targetEntity']);
+            $association    = $targetEntity->associationMappings[$mapping['mappedBy']];
+        }
+
+        $joinColumns = ( ! $mapping['isOwningSide'])
+            ? $association['joinTable']['inverseJoinColumns']
+            : $association['joinTable']['joinColumns'];
+
+        foreach ($joinColumns as $joinColumn) {
+            $columnName     = $this->quoteStrategy->getJoinColumnName($joinColumn, $class, $this->platform);
+            $referencedName = $joinColumn['referencedColumnName'];
+            $conditions[]   = 't.' . $columnName . ' = ?';
+            $params[]       = ($class->containsForeignIdentifier)
+                ? $id[$class->getFieldForColumn($referencedName)]
+                : $id[$class->fieldNames[$referencedName]];
+        }
+
+        $joinTableName = $this->quoteStrategy->getJoinTableName($association, $class, $this->platform);
+        list($joinTargetEntitySQL, $filterSql) = $this->getFilterSql($mapping);
+
+        if ($filterSql) {
+            $conditions[] = $filterSql;
+        }
+
+        $sql = 'SELECT COUNT(*)'
```

Comment:
Is there a cross-platform compliant way to avoid `COUNT(*)` and use `COUNT(1)` or such instead?